### PR TITLE
Add LanguageVersion for C# 13

### PR DESCRIPTION
**Pull request by jcouv**
Pinged Lee and Rahul by email to update releases.json

### CODE REVIEW

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Portable/LanguageVersion.cs [hunk 1]
```diff
@@ -231,6 +231,22 @@ public enum LanguageVersion
...
+        /// <br/>
+        /// Features:
+        /// <list type="bullet">
+        /// <item><description>Escape character</description></item>
+        /// <item><description>Method group natural type improvements</description></item>
+        /// <item><description>`Lock` object</description></item>
+        /// <item><description>Implicit indexer access in object initializers</description></item>
+        /// <item><description>`params` collections</description></item>
+        /// <item><description>ref/unsafe in iterators/async</description></item>
+        /// <item><description>`allows ref struct` constraint</description></item>
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/LanguageVersion.cs:245]:
```suggestion
        /// <item><description>`allows ref struct` constraint</description></item>
        /// <item><description>Partial properties</description></item>
```

File: src/Workspaces/CSharpTest/EmbeddedLanguages/VirtualChars/CSharpVirtualCharServiceTests.cs [hunk 2]
```diff
@@ -164,11 +164,7 @@ public void TestEscapeInInterpolatedSimpleString()
 
         [Fact]
         public void TestEscapeInInterpolatedSimpleString_E()
-            => TestFailure("$\"\\e\"");
-
-        [Fact]
-        public void TestEscapeInInterpolatedSimpleString_E_Preview()
-            => Test("$\"\\e\"", @"['\u001B',[2,4]]", CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+            => Test("$\"\\e\"", @"['\u001B',[2,4]]");
```
Comment (RikkiGibson) [inline src/Workspaces/CSharpTest/EmbeddedLanguages/VirtualChars/CSharpVirtualCharServiceTests.cs:167]: It feels reasonable to update these tests by using LanguageVersion.CSharp12 for the failure case and CSharp13 for the success case, and renaming the tests accordingly

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by SystematicChaos012** [comment]
Does this mean that the remaining features will be updated in C# 14?

**Comment by jcouv** [comment]
@SystematicChaos012 No, there are a few more feature that we're still working on but expect to make it into C# 13.

**Pull request merged**